File: core4/leetcode/Arrays/P682/main.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int calPoints(const vector<string>& operations) {
        vector<int> result;

        for (const auto& op : operations) {
            if (op == "C") {
                if (!result.empty()) result.pop_back(); // remove last element
            }
            else if (op == "D") {
                result.push_back(2 * result.back()); // double last score
            }
            else if (op == "+") {
                int last1 = result.back();
                int last2 = result[result.size() - 2];
                result.push_back(last1 + last2); // sum last two scores
            }
            else {
                result.push_back(stoi(op)); // convert string to int and add
            }
        }

        int sum = 0;
        for (auto score : result) sum += score; // sum all scores
        return sum;
    }
};

int main() {
    vector<string> operations = {"5","2","C","D","+"};
    Solution obj;
    cout << "Sum of operations: " << obj.calPoints(operations);
}

File: core4/leetcode/Arrays/P682/main.py
class Solution:
    def calPoints(self,operations):
        if not operations:
            return 0
           
        result=[]
       
        for i in range(len(operations)):
            if operations[i] == 'C':
                result.pop()
            elif operations[i] == 'D':
                last = result[-1]
                result.append(2*last)
            elif operations[i] == "+":
                result.append(result[-1]+result[-2])
            else:
                result.append(int(operations[i]))  
               
        sumation=0
        for i in range(len(result)):
            sumation += result[i]
       
        return sumation
               
               
               
               


# 🔽 Main
if __name__ == "__main__":
    ops = ["5","2","C","D","+"]
   
    sol = Solution()
    print("Total Score:", sol.calPoints(ops))

